File: packages/shared-contracts/src/blueprint.ts
export const BLUEPRINT_QUEST_DIFFICULTY_VALUES = ['low', 'medium', 'high'] as const;
export const BLUEPRINT_QUEST_STATUS_VALUES = ['PENDING', 'IN_PROGRESS', 'COMPLETED'] as const;

export type BlueprintQuestDifficulty = (typeof BLUEPRINT_QUEST_DIFFICULTY_VALUES)[number];
export type BlueprintQuestStatus = (typeof BLUEPRINT_QUEST_STATUS_VALUES)[number];

export interface BlueprintQuest {
  id: string;
  subject: string;
  title: string;
  summary: string;
  difficulty: BlueprintQuestDifficulty;
  why_this_matters: string;
  expected_record_impact: string;
  recommended_output_type: string;
  status: BlueprintQuestStatus;
}

export interface BlueprintQuestGroup {
  name: string;
  quests: BlueprintQuest[];
}

export interface CurrentBlueprintResponse {
  id: string;
  project_id: string;
  project_title: string;
  target_major: string | null;
  headline: string;
  recommended_focus: string;
  semester_priority_message: string;
  priority_quests: BlueprintQuest[];
  subject_groups: BlueprintQuestGroup[];
  activity_groups: BlueprintQuestGroup[];
  expected_record_effects: string[];
  created_at: string;
}

export interface QuestStarterChoice {
  id: string;
  label: string;
  prompt: string;
}

export interface QuestStartPayload {
  quest_id: string;
  blueprint_id: string;
  project_id: string;
  project_title: string;
  target_major: string | null;
  subject: string;
  title: string;
  summary: string;
  why_this_matters: string;
  expected_record_impact: string;
  recommended_output_type: string;
  status: BlueprintQuestStatus;
  workshop_intro: string;
  document_seed_markdown: string;
  starter_choices_seed: QuestStarterChoice[];
}

File: backend/services/api/src/unifoli_api/schemas/blueprint.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

BlueprintQuestDifficulty = Literal["low", "medium", "high"]
BlueprintQuestStatus = Literal["PENDING", "IN_PROGRESS", "COMPLETED"]


class QuestRead(BaseModel):
    id: str
    subject: str
    title: str
    summary: str
    difficulty: BlueprintQuestDifficulty
    why_this_matters: str
    expected_record_impact: str
    recommended_output_type: str
    status: BlueprintQuestStatus


class QuestGroupRead(BaseModel):
    name: str
    quests: list[QuestRead]


class CurrentBlueprintRead(BaseModel):
    id: str
    project_id: str
    project_title: str
    target_major: str | None = None
    headline: str
    recommended_focus: str
    semester_priority_message: str
    priority_quests: list[QuestRead]
    subject_groups: list[QuestGroupRead]
    activity_groups: list[QuestGroupRead]
    expected_record_effects: list[str]
    created_at: datetime


class StarterChoiceRead(BaseModel):
    id: str
    label: str
    prompt: str


class QuestStartResponse(BaseModel):
    quest_id: str
    blueprint_id: str
    project_id: str
    project_title: str
    target_major: str | None = None
    subject: str
    title: str
    summary: str
    why_this_matters: str
    expected_record_impact: str
    recommended_output_type: str
    status: BlueprintQuestStatus
    workshop_intro: str
    document_seed_markdown: str
    starter_choices_seed: list[StarterChoiceRead]
